### PR TITLE
Updated to dexlib2 and fixed excludedAnnotation problem

### PR DESCRIPTION
**Pull request by Bersh**
Hi maintainers! 
I have updated `dexlib` version to `2.4.0` to fix build error described here: https://github.com/shazam/fork/issues/177
Also I have changed `excludedAnnotation` behaviour, now tests with excludedAnnotation is considered ignored. This is aimed to fix this issue: https://github.com/shazam/fork/issues/169

Please let me know what do you think about those changes

### CODE REVIEW

File: fork-common-test-dex/app/src/androidTest/java/com/shazam/forktest/ClassWithExcludedAnnotationTest.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        assertEquals(4, 2 + 2);
+    }
+
+    @Test
+    @ExcludedAnnotation
+    public void excludedMethod() {
+        assertEquals(4, 2 + 2);
+    }
+
+}
```
Comment (silviul) [inline fork-common-test-dex/app/src/androidTest/java/com/shazam/forktest/ClassWithExcludedAnnotationTest.java:20]: style: can we add an empty line at the end of the file?

**Review by silviul — CHANGES_REQUESTED** [review]

File: fork-common/src/test/java/com/shazam/fork/suite/TestSuiteLoaderTest.java [hunk 2]
```diff
@@ -83,9 +97,31 @@ private DexFile testDexFile() {
...
+    @SuppressWarnings("unchecked")
+    @Test
+    public void setsExcludedAnnotationFlag() throws Exception {
+        assertThat(testSuiteLoader.loadTestSuite(), hasItems(
+                sameTestEventAs("includedMethod", "com.shazam.forktest.ClassWithExcludedAnnotationTest", false),
+                sameTestEventAs("excludedMethod", "com.shazam.forktest.ClassWithExcludedAnnotationTest", true)));
     }
 
+    @Test
+    public void EmptyExcludedAnnotationWorks() throws Exception {
```
Comment (silviul) [inline fork-common/src/test/java/com/shazam/fork/suite/TestSuiteLoaderTest.java:112]: can we also name this method starting with a lowercase "e"?

**Review by silviul — CHANGES_REQUESTED** [review]

File: fork-common-test-dex/app/src/androidTest/java/com/shazam/forktest/ClassWithExcludedAnnotationTest.java (hunk 1 above)
Comment (silviul) [reply fork-common-test-dex/app/src/androidTest/java/com/shazam/forktest/ClassWithExcludedAnnotationTest.java:20]: also no space before the final curly?

Comment (Bersh) [reply fork-common-test-dex/app/src/androidTest/java/com/shazam/forktest/ClassWithExcludedAnnotationTest.java]: should be find after d6bfea2

**Review by silviul — APPROVED** [review]

**Pull request merged**